Fail to launch

```
FATAL EXCEPTION: main
Process: cc.wordview.app, PID: 18461
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:566)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1038)
Caused by: java.lang.reflect.InvocationTargetException
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:556)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1038) 
Caused by: javax.crypto.AEADBadTagException
	at android.security.keystore2.AndroidKeyStoreCipherSpiBase.engineDoFinal(AndroidKeyStoreCipherSpiBase.java:611)
	at javax.crypto.Cipher.doFinal(Cipher.java:2132)
	at com.google.crypto.tink.integration.android.AndroidKeystoreAesGcm.decryptInternal(AndroidKeystoreAesGcm.java:118)
	at com.google.crypto.tink.integration.android.AndroidKeystoreAesGcm.decrypt(AndroidKeystoreAesGcm.java:101)
	at com.google.crypto.tink.KeysetHandle.decrypt(KeysetHandle.java:919)
	at com.google.crypto.tink.KeysetHandle.readWithAssociatedData(KeysetHandle.java:804)
	at com.google.crypto.tink.KeysetHandle.read(KeysetHandle.java:785)
	at com.google.crypto.tink.integration.android.AndroidKeysetManager$Builder.readMasterkeyDecryptAndParseKeyset(AndroidKeysetManager.java:381)
	at com.google.crypto.tink.integration.android.AndroidKeysetManager$Builder.build(AndroidKeysetManager.java:297)
	at androidx.security.crypto.EncryptedSharedPreferences.create(EncryptedSharedPreferences.java:169)
	at cc.wordview.app.api.JwtSaverKt.getStoredJwt(JwtSaver.kt:48)
	at cc.wordview.app.ui.activities.home.composables.home.HomeKt.Home(Home.kt:51)
	at cc.wordview.app.ui.activities.home.HomeNav$Home.Composable(HomeNav.kt:46)
	at cc.wordview.app.ui.activities.home.ComposableSingletons$HomeActivityKt$lambda-1$1$1$1$1.invoke(HomeActivity.kt:45)
	at cc.wordview.app.ui.activities.home.ComposableSingletons$HomeActivityKt$lambda-1$1$1$1$1.invoke(HomeActivity.kt:44)
	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:142)
	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:51)
	at androidx.navigation.compose.NavHostKt$NavHost$32$1.invoke(NavHost.kt:703)
	at androidx.navigation.compose.NavHostKt$NavHost$32$1.invoke(NavHost.kt:702)
	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:121)
	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:51)
	at androidx.compose.runtime.CompositionLocalKt.CompositionLocalProvider(CompositionLocal.kt:384)
	at androidx.compose.runtime.saveable.SaveableStateHolderImpl.SaveableStateProvider(SaveableStateHolder.kt:79)
	at androidx.navigation.compose.NavBackStackEntryProviderKt.SaveableStateProvider(NavBackStackEntryProvider.kt:69)
	at androidx.navigation.compose.NavBackStackEntryProviderKt.access$SaveableStateProvider(NavBackStackEntryProvider.kt:1)
	at androidx.navigation.compose.NavBackStackEntryProviderKt$LocalOwnersProvider$1.invoke(NavBackStackEntryProvider.kt:56)
	at androidx.navigation.compose.NavBackStackEntryProviderKt$LocalOwnersProvider$1.invoke(NavBackStackEntryProvider.kt:55)
	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:121)
	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:51)
	at androidx.compose.runtime.CompositionLocalKt.CompositionLocalProvider(CompositionLocal.kt:364)
	at androidx.navigation.compose.NavBackStackEntryProviderKt.LocalOwnersProvider(NavBackStackEntryProvider.kt:51)
	at androidx.navigation.compose.NavHostKt$NavHost$32.invoke(NavHost.kt:702)
	at androidx.navigation.compose.NavHostKt$NavHost$32.invoke(NavHost.kt:682)
	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:142)
	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:51)
	at androidx.compose.animation.AnimatedContentKt$AnimatedContent$6$1$5.invoke(AnimatedContent.kt:862)

```

Probably due to deprecated functions used in JwtSaver.kt

--- a/app/src/main/java/cc/wordview/app/api/JwtSaver.kt
+++ b/app/src/main/java/cc/wordview/app/api/JwtSaver.kt
@@ -21,53 +21,60 @@ import android.content.Context
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.platform.LocalContext
 import androidx.security.crypto.EncryptedSharedPreferences
-import androidx.security.crypto.MasterKeys
+import androidx.security.crypto.MasterKey
 import androidx.core.content.edit
+import java.io.IOException
+import javax.crypto.AEADBadTagException
 
-@Suppress("DEPRECATION")
-val masterKeyAlias = MasterKeys.getOrCreate(MasterKeys.AES256_GCM_SPEC)
-
+private const val PREF_NAME = "user_store"
 private const val TOKEN_KEY = "jwt_token"
 
-fun getStoredJwt(context: Context): String? {
-    val sharedPreferences = EncryptedSharedPreferences.create(
-        "user_store",
-        masterKeyAlias,
+private fun getMasterKey(context: Context): MasterKey =
+    MasterKey.Builder(context)
+        .setKeyScheme(MasterKey.KeyScheme.AES256_GCM)
+        .build()
+
+private fun getEncryptedPrefs(context: Context) =
+    EncryptedSharedPreferences.create(
         context,
+        PREF_NAME,
+        getMasterKey(context),
         EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
         EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
     )
 
-    return sharedPreferences.getString(TOKEN_KEY, null)
+fun getStoredJwt(context: Context): String? {
+    return try {
+        val sharedPreferences = getEncryptedPrefs(context)
+        sharedPreferences.getString(TOKEN_KEY, null)
+    } catch (e: AEADBadTagException) {
+        context.deleteSharedPreferences(PREF_NAME)
+        null
+    } catch (e: IOException) {
+        context.deleteSharedPreferences(PREF_NAME)
+        null
+    }
 }
 
 @Composable
 fun getStoredJwt(): String? {
     val context = LocalContext.current
 
-    val sharedPreferences = EncryptedSharedPreferences.create(
-        "user_store",
-        masterKeyAlias,
-        context,
-        EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
-        EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
-    )
-
-    return sharedPreferences.getString(TOKEN_KEY, null)
+    return try {
+        val sharedPreferences = getEncryptedPrefs(context)
+        sharedPreferences.getString(TOKEN_KEY, null)
+    } catch (e: AEADBadTagException) {
+        context.deleteSharedPreferences(PREF_NAME)
+        null
+    } catch (e: IOException) {
+        context.deleteSharedPreferences(PREF_NAME)
+        null
+    }
 }
 
 fun setStoredJwt(token: String, context: Context) {
-    val sharedPreferences = EncryptedSharedPreferences.create(
-        "user_store",
-        masterKeyAlias,
-        context,
-        EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
-        EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
-    )
-
+    val sharedPreferences = getEncryptedPrefs(context)
     sharedPreferences.edit(commit = true) {
         putString(TOKEN_KEY, token)
     }
 }
-
-

